Deprecation related to Polars 1.32's new Categorical/Enum implementation

When running `import pandera.polars as pa` with polars 1.32 installed, the below deprecation warning is raised:

```none
    import pandera.polars as pa
..\.venv\Lib\site-packages\pandera\polars.py:11: in <module>
    from pandera.api.polars.components import Column
..\.venv\Lib\site-packages\pandera\api\polars\components.py:15: in <module>
    from pandera.engines import polars_engine
..\.venv\Lib\site-packages\pandera\engines\polars_engine.py:730: in <module>
    @Engine.register_dtype(equivalents=[pl.Categorical])
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
..\.venv\Lib\site-packages\pandera\engines\engine.py:217: in _wrapper
    cls._register_equivalents(pandera_dtype_cls, *equivalents)
..\.venv\Lib\site-packages\pandera\engines\engine.py:160: in _register_equivalents
    pandera_dtype = pandera_dtype_cls()  # type: ignore
                    ^^^^^^^^^^^^^^^^^^^
..\.venv\Lib\site-packages\pandera\engines\polars_engine.py:744: in __init__
    object.__setattr__(self, "type", pl.Categorical(ordering=ordering))
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
..\.venv\Lib\site-packages\polars\datatypes\classes.py:796: in __init__
    issue_deprecation_warning(
..\.venv\Lib\site-packages\polars\_utils\deprecation.py:61: in issue_deprecation_warning
    issue_warning(message, DeprecationWarning)
..\.venv\Lib\site-packages\polars\_utils\various.py:494: in issue_warning
    warnings.warn(
E   DeprecationWarning: the physical Categorical ordering is deprecated. The ordering is now always lexical.
E   (Deprecated in version 1.32.0)
```
This is related to https://github.com/pola-rs/polars/pull/23016.

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of pandera.
- [x] (optional) I have confirmed this bug exists on the main branch of pandera.

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
# pip install "polars>=1.32" --upgrade
import pandera.polars as pa
```

#### Expected behavior
The import should succeed without deprecations.

--- a/pandera/engines/polars_engine.py
+++ b/pandera/engines/polars_engine.py
@@ -724,24 +724,18 @@ class Categorical(DataType):
 
     type = pl.Categorical
 
-    ordering = None
-
-    def __init__(
-        self,
-        ordering: Literal["physical", "lexical"] | None = "lexical",
-    ) -> None:
-        object.__setattr__(self, "ordering", ordering)
+    def __init__(self) -> None:
         object.__setattr__(self, "type", pl.Categorical())
 
     def __deepcopy__(self, memo):
         """Custom deepcopy to avoid pickling issues with pl.Categorical()."""
-        return self.__class__(ordering=self.ordering)
+        return self.__class__()
 
     @classmethod
     def from_parametrized_dtype(cls, polars_dtype: pl.Categorical):
         """Convert a :class:`polars.Categorical` to
         a Pandera :class:`pandera.engines.polars_engine.Categorical`."""
-        return cls(ordering=polars_dtype.ordering)
+        return cls()
 
 
 @Engine.register_dtype(equivalents=[pl.Enum])

--- a/tests/polars/test_polars_dtypes.py
+++ b/tests/polars/test_polars_dtypes.py
@@ -540,3 +540,28 @@ def test_datetime_time_zone_agnostic(dtype):
         dtype.type, "time_zone", None
     ):
         assert not tz_sensitive_utc.check(dtype)
+
+
+def test_polars_import_no_warnings():
+    """Regression test for https://github.com/unionai-oss/pandera/issues/2104.
+
+    Importing pandera.polars should not emit DeprecationWarning or
+    FutureWarning related to polars Categorical/Enum changes.
+    """
+    import subprocess
+    import sys
+
+    result = subprocess.run(
+        [
+            sys.executable,
+            "-W",
+            "error::DeprecationWarning",
+            "-W",
+            "error::FutureWarning",
+            "-c",
+            "import pandera.polars",
+        ],
+        capture_output=True,
+        text=True,
+    )
+    assert result.returncode == 0, f"Import raised a warning:\n{result.stderr}"

